judge/management/commands/generate_report.py

```diff
@@ -0,0 +1,95 @@
+import os
+import shutil
+import sys
+
+from django.conf import settings
+from django.core.mail import EmailMessage
+from django.core.management.base import BaseCommand
+from django.template.loader import get_template
+
+from judge.models import Contest, ContestParticipation
+from judge.pdf_problems import DefaultPdfMaker, PhantomJSPdfMaker, PuppeteerPDFRender, SeleniumPDFRender, \
+    SlimerJSPdfMaker
+
+
+def generate_scoreboard(contest, period):
+    problem_ids = [str(prob.id) for prob in contest.contest_problems.all().order_by('order')]
+    out = ''
+
+    for registrant in contest.registrants.all().order_by('user__user__last_name'):
+        if period and int(registrant.data['class-period']) != period:
```

Comment:
```suggestion
        if period is not None and int(registrant.data['class-period']) != period:
```
In case `period` is `0`.